lib/tasks/bounty_tasks.rake

```diff
@@ -0,0 +1,10 @@
+desc 'send open bounties email'
+task send_bounty_email: :environment do
+  @bounties = Bounty.open.last(15)
+  if @bounties.any?
+    @users = User.all
```

Comment:
Hi @clair13, 
I feel this is performance intensive task (although it is rake task but still) as it will load the entire table into memory, instantiating all the rows; then iterate through the instances.

We can use `find_each`, as it does this in batches, which is more efficient in terms of memory usage.

something like below
```ruby
User.find_each do |user|
  BountyMailer.with(user: user, bounties: @bounties).open_bounty.deliver_later
end
```

References
[Retrieving Multiple Objects in Batches](https://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches)
[ActiveRecord::Batches](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html)

Reply:
Thank you for the suggestion,I fixed it based on your suggestion.